### PR TITLE
DEV: updates chart.js to 3.5.1

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: plugins/poll/assets/javascripts/components/poll-breakdown-chart.js.es6 [hunk 1]
```diff
@@ -122,10 +124,14 @@ export default Component.extend({
         responsive: true,
         aspectRatio: 1.1,
         animation: { duration: 0 },
-        tooltips: false,
+
```
Comment (jjaffeux) [inline plugins/poll/assets/javascripts/components/poll-breakdown-chart.js.es6:127]: This code was causing lots of ember error

File: plugins/poll/assets/javascripts/controllers/poll-breakdown.js.es6 [hunk 2]
```diff
@@ -47,7 +47,6 @@ export default Controller.extend(ModalFunctionality, {
     loadScript("/javascripts/Chart.min.js")
       .then(() => loadScript("/javascripts/chartjs-plugin-datalabels.min.js"))
       .then(() => {
-        window.Chart.plugins.unregister(window.ChartDataLabels);
```
Comment (jjaffeux) [inline plugins/poll/assets/javascripts/controllers/poll-breakdown.js.es6:50]: This is not needed anymore in de datalabels version as it's not registered by default

File: plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6 [hunk 3]
```diff
@@ -465,17 +465,19 @@ createWidget("discourse-poll-pie-canvas", {
...
-      let chart = new Chart(el.getContext("2d"), config);
-      document.getElementById(
-        `poll-results-legend-${attrs.id}`
-      ).innerHTML = chart.generateLegend();
+      this._chart = new Chart(el.getContext("2d"), config);
     });
   },
 
+  willRerenderWidget() {
+    this._chart?.destroy();
```
Comment (jjaffeux) [inline plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6:478]: we were not reseting the canvas which should be done, and is fatal error in 3.x anyways

File: plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6 [hunk 4]
```diff
@@ -497,15 +499,54 @@ createWidget("discourse-poll-pie-chart", {
...
     contents.push(chart);
 
-    contents.push(h(`div#poll-results-legend-${attrs.id}.pie-chart-legends`));
+    contents.push(h(`ul#poll-results-legend-${attrs.id}.pie-chart-legends`));
 
     return contents;
   },
 });
 
+const htmlLegendPlugin = {
```
Comment (jjaffeux) [inline plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6:508]: generaleLegend doesn't exist anymore in 3.X so writing a custom plugin

**Comment by jjaffeux** [comment]
Besides updating chart.js/datalabels and fixing various bugs. Switching to a custom legend plugin allows us to be able to click legend, this feature was lost before.

**Comment by jjaffeux** [comment]
Failing test in data-explorer will be fixed by:

https://github.com/discourse/discourse-data-explorer/pull/132

**Review by CvX — APPROVED** [review]

**Pull request merged**